lib/event_listeners.js

```diff
@@ -211,6 +211,25 @@ AWS.EventListeners = {
       req.httpRequest.headers['Host'] = req.httpRequest.endpoint.host;
     });
 
+    add('SET_TRACE_ID', 'afterBuild', function SET_TRACE_ID(req) {
+      if (AWS.util.isNode()) {
+        var ENV_LAMBDA_FUNCTION_NAME = 'AWS_LAMBDA_FUNCTION_NAME';
+        var ENV_TRACE_ID = '_X_AMZN_TRACE_ID';
+        var traceIdHeaderName = 'X-Amzn-Trace-Id';
+        var functionName = process.env[ENV_LAMBDA_FUNCTION_NAME];
+        var traceId = process.env[ENV_TRACE_ID];
+        if (
+          typeof functionName === 'string' &&
+          functionName.length > 0 &&
+          typeof traceId === 'string' &&
+          traceId.length > 0 &&
+          !Object.hasOwnProperty.call(req.httpRequest.headers, traceIdHeaderName)
+        ) {
+          req.httpRequest.headers[traceIdHeaderName] = traceId;
```

Comment:
The traceId should be precent-encoded as per [RFC 3986-2.1](https://datatracker.ietf.org/doc/html/rfc3986#section-2.1).

For example, `first\nsecond` should be encoded to `first%0Asecond`

Reply:
The [RFC 3986-2.1](https://datatracker.ietf.org/doc/html/rfc3986#section-2.1) is for URI, a header is not URI. Header doesn't require encoding per [rfc2616-4.2](https://datatracker.ietf.org/doc/html/rfc2616#section-4.2)